client/llb/source.go

```diff
@@ -287,6 +297,19 @@ func AuthHeaderSecret(v string) GitOption {
 	})
 }
 
+func KnownSSHHosts(key string) GitOption {
+	key = strings.TrimSuffix(key, "\n")
+	return gitOptionFunc(func(gi *GitInfo) {
+		gi.KnownSSHHosts = gi.KnownSSHHosts + key + "\n"
+	})
+}
+
+func MountSSHSock() GitOption {
+	return gitOptionFunc(func(gi *GitInfo) {
+		gi.MountSSHSock = true
```

Comment:
This should be the ssh id, not a bool. If identifier is SSH based it should be set automatically.

Reply:
great point, changed this to a string which is now passed from the client.

Reply:
>  If identifier is SSH based it should be set automatically.

That part does not look addressed yet.

Reply:
Are you sure about that Tonis? I made that change under https://github.com/moby/buildkit/pull/1782/files#diff-72973099cb7ef03a01202ce533dab47469e238d3fed58286194e20c015d9199fR307 perhaps you're looking at an outdated version?

Reply:
I missed that part, i'm working on implementing it as well as a ssh keyscan for when no keys are supplied.